src/main/java/hudson/plugins/ws_cleanup/WsCleanup.java

```diff
@@ -10,59 +10,105 @@
 import hudson.matrix.MatrixProject;
 import hudson.model.BuildListener;
 import hudson.model.Result;
-import hudson.model.AbstractBuild;
+import hudson.model.Run;
+import hudson.model.TaskListener;
 import hudson.tasks.BuildStepDescriptor;
 import hudson.tasks.BuildStepMonitor;
 import hudson.tasks.Notifier;
 import hudson.tasks.Publisher;
-import hudson.slaves.EnvironmentVariablesNodeProperty;
+import jenkins.tasks.SimpleBuildStep;
+import net.sf.json.JSONObject;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
 
 import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import net.sf.json.JSONObject;
-
-import org.kohsuke.stapler.DataBoundConstructor;
-import org.kohsuke.stapler.StaplerRequest;
-
 /**
  *
  * @author dvrzalik
  */
-public class WsCleanup extends Notifier implements MatrixAggregatable {
+public class WsCleanup extends Notifier implements MatrixAggregatable, SimpleBuildStep {
 
     public static final String LOG_PREFIX = "[WS-CLEANUP] ";
 
-    private final List<Pattern> patterns;
-    private final boolean deleteDirs;
+    private List<Pattern> patterns = Collections.emptyList();
+    private boolean deleteDirs = false;
 
     @Deprecated
-    private boolean skipWhenFailed; // keep it for backward compatibility
-    private boolean cleanWhenSuccess;
-    private boolean cleanWhenUnstable;
-    private boolean cleanWhenFailure;
-    private boolean cleanWhenNotBuilt;
-    private boolean cleanWhenAborted;
+    private boolean skipWhenFailed = false; // keep it for backward compatibility
+    private boolean cleanWhenSuccess = true;
+    private boolean cleanWhenUnstable = true;
+    private boolean cleanWhenFailure = true;
+    private boolean cleanWhenNotBuilt = true;
+    private boolean cleanWhenAborted = true;
```

Comment:
In the *WsCleanup/config.jelly* they have `default="true"`, so I think it makes sense to match the jelly file.